dqvm/src/main.lisp

```diff
@@ -73,25 +72,26 @@
   (when (master-node-p)
     (show-welcome))
 
-  (when execute
-    ;; Do all file handling here, not in the node code.
-    (when (master-node-p)
-      (setf *program* (safely-read-quil execute)))
+  (let (program)
+    (when execute
+      ;; Do all file handling here, not in the node code.
+      (when (master-node-p)
+        (setf program (safely-read-quil execute)))
 
-    (assert (and (plusp +worker-count+)
-                 (power-of-two-p +worker-count+))
-            ()
-            "There has to be a positive power-of-two ~
+      (assert (and (plusp +worker-count+)
+                   (power-of-two-p +worker-count+))
+              ()
+              "There has to be a positive power-of-two ~
              number of workers. We have ~D."
-            +worker-count+)
-    (with-errors-printed-verbosely
-      (unwind-protect
-           (if (master-node-p)
-               (dqvm.master:%main-master :program *program*)
-               (dqvm.worker:%main-worker))
-        (mpi:mpi-finalize)))
-
-    (quit-nicely)))
+              +worker-count+)
+      (with-errors-printed-verbosely
+          (unwind-protect
+               (if (master-node-p)
+                   (dqvm.master:%main-master :program program)
```

Comment:
Personally, I would just nix the `program` binding and directly pass `:program (safely-read-quil execute)` here, unless there is some reason it needs to happen outside the `with-errors-printed-verbosely` / `unwind-protect`.